codewof/programming/content/en/while-away/question.md

```diff
@@ -0,0 +1,7 @@
+# While Away
+
+The following function `while_away(minutes)` is supposed to take a number of minutes and **print** the message `I'm away` for each minute until the number of minutes runs out, and then **print** `I'm back`.
```

Comment:
I like this question a lot! But I think it's not quite clear enough that the program is meant to be counting down. I think users might read this question and think that they have to try and write a program that uses a time function to actually wait for that amount of time or something like that

Reply:
Yeah I would agree with this, at least that is how I read it

Reply:
I've changed it to 

> The following function `while_away(minutes)` is supposed to take a number of minutes and **print** the message `I'm away` for each minute, and then **print** `I'm back`.

Do you think that works better?

Reply:
I still read it the same way tbh. Maybe add this to the end of it: "To be clear, your function does **not** need to wait for a minute between printing each message. It only needs to modify the number of minutes every time the loop runs so that the loop condition can be satisfied."

I think it is fine that this extra message would kind of tell them the answer, because gaining exposure to the problem (infinite while loop) and the solution (make sure the loop condition can be satisfied) is more important.

Reply:
Sounds good! Would it be ok if instead of saying they need to modify the number of minutes it says "The loop just needs to repeat for the same number of times as the number of minutes", so it gives it away a little bit less? I'm happy with your version though if that doesn't sound good to you :)

Reply:
Yep, that's fine *thumbs up*

Reply:
:D